src/components/MessagesList/MessagesList.vue

```diff
@@ -1217,6 +1218,11 @@ export default {
 		messagesGroupComponent(group) {
 			return group.isSystemMessagesGroup ? MessagesSystemGroup : MessagesGroup
 		},
+
+		onMessageHeightChanged({ heightDiff }) {
+			// scroll down by the height difference
+			this.$refs.scroller.scrollTop += heightDiff
```

Comment:
That looks better for me, but we need to track scrolling direction (could be done with `onScroll` event, and `this.previousScrollTopValue`), and it may be still flacky on corner cases (conversation switch, initial rendering)
 
```suggestion
			if (this.lastScrollDirection === 'up' || this.isChatScrolledToBottom) {
				// scroll down by the height difference
				this.$refs.scroller.scrollTop += heightDiff
			}
```

Reply:
I don't see a reason to specify the direction as jumping happens anyway.

Reply:
Try this one. Feels less jumpy for me, when scrolling in both directions
```suggestion
			// scroll down by the height difference
			this.$refs.scroller.scrollTop += this.isChatScrolledToBottom ? heightDiff : heightDiff / 2
```

Reply:
why divided by 2 when it's not at the bottom ? it is always scrolled by heightDiff

Reply:
If scroll to the middle of the widget, it looks like it's expanding in both directions